tests/test_models/test_catboost.py

```diff
@@ -144,3 +144,79 @@ def test_save_load(model, example_tsds):
     horizon = 3
     transforms = [LagTransform(in_column="target", lags=list(range(horizon, horizon + 3)))]
     assert_model_equals_loaded_original(model=model, ts=example_tsds, transforms=transforms, horizon=horizon)
+
+
+@pytest.fixture()
+def ar_dataset_w_exog():
+    df = generate_ar_df(start_time="2021-01-01", periods=100, n_segments=2)
+    df_exog = df.copy()
+    df_exog["f1"] = np.sin(df_exog["target"])
+    df_exog["f2"] = np.cos(df_exog["target"])
+    df_exog.drop(columns=["target"], inplace=True)
+
+    df = TSDataset.to_dataset(df)
+    df_exog = TSDataset.to_dataset(df_exog)
+
+    ts = TSDataset(df=df, df_exog=df_exog, freq="D")
+    return ts
+
+
+def test_forecast_prediction_components_equal(ar_dataset_w_exog):
+    train, test = ar_dataset_w_exog.train_test_split(test_size=5)
```

Comment:
I guess to simplify logic we can train and predict components on the same dataset, we don't actually need to generate future(relevant for all the tests)

Reply:
This is to keep consistent with `forecast` flow and to make tests obvious. Also, using future we can decompose only on a small portion of data instead of running on the whole training set.

Reply:
We can make train dataset small
Here we believe that forecast flow works perfect and in the `future` we get the dataset in correct format, so here we just check that `predict_components works`.